src/services/users/update-user.js

```diff
@@ -0,0 +1,64 @@
+const db = require("../../db")
+const customError = require("../../lib/custom-error")
+const { UserNotFoundError } = require("../errors")
+
+/**
+ * 
+ * @param {Number} id - id of the user to be updated
+ * @param {Object} cols - columns to be updated
+ * @returns {String} - query to be executed
+ */
+const constructQuery = (id, cols) => {
+
+	// Setup static beginning of query
+	const query = ['UPDATE users']
+	query.push('SET')
+  
+	// Create another array storing each set command
+	// and assigning a number value for parameterized query
+	const set = []
+	Object.keys(cols).forEach((key, i) => {
+	  set.push(`"${key}" = ($${(i + 1)})`) 
+	})
+	query.push(set.join(', '))
+  
+	// Add the WHERE statement to look up by id
+	query.push(`WHERE id = ${id} RETURNING  "id",
```

Comment:
so though the answer is a great starting point, we will still need to not pass the `id` as a parameterised value (the dollar sign thing) to still get the benefit of preventing sql injection

we can count the `set` array to know what number to assign to the `id` and then attach id as the last value to the `colValues`

or any other approach that you might find useful. The answer I shared has a comment that talks about this